java/flight/src/main/java/org/apache/arrow/flight/CallStatus.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.flight;
+
+import java.util.Objects;
+
+/**
+ * The result of a Flight RPC.
+ */
+public class CallStatus {
+
+  private final FlightStatusCode code;
+  private final Throwable cause;
+  private final String description;
+
+  public static final CallStatus OK = FlightStatusCode.OK.toStatus();
+  public static final CallStatus UNKNOWN = FlightStatusCode.UNKNOWN.toStatus();
+  public static final CallStatus INTERNAL = FlightStatusCode.INTERNAL.toStatus();
+  public static final CallStatus INVALID_ARGUMENT = FlightStatusCode.INVALID_ARGUMENT.toStatus();
+  public static final CallStatus TIMED_OUT = FlightStatusCode.TIMED_OUT.toStatus();
+  public static final CallStatus CANCELLED = FlightStatusCode.CANCELLED.toStatus();
+  public static final CallStatus UNAUTHENTICATED = FlightStatusCode.UNAUTHENTICATED.toStatus();
+  public static final CallStatus UNAUTHORIZED = FlightStatusCode.UNAUTHORIZED.toStatus();
+  public static final CallStatus UNIMPLEMENTED = FlightStatusCode.UNIMPLEMENTED.toStatus();
+  public static final CallStatus UNAVAILABLE = FlightStatusCode.UNAVAILABLE.toStatus();
+
+  /**
+   * Create a new status.
+   * @param code The status code.
+   * @param cause An exception that resulted in this status (or null).
+   * @param description A description of the status (or null).
+   */
+  public CallStatus(FlightStatusCode code, Throwable cause, String description) {
```

Comment:
consider making a construct that just take code, so you don't have to make nullable parameters for constructing from the the enum.